Newsletter: What nobody tells developers about documentation

ivanagas opened the pull request:
## Changes

Feedback appreciated:

High level feedback appreciated:

- Is this opinionated enough?
- Any better examples or points to include?

## Checklist

- [x] Words are spelled using American English
- [x] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [x] Feature names are in **[sentence case too]([https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case))**
- [x] Use relative URLs for internal links

## Article checklist

- [x] I've checked the preview build of the article
- [x] The date on the article is today's date

andyvan-ph reviewed (commented):
I don't love this, tbh. Reasons why:

- It all feels a little high-level and and obvious. Nothing in it surprised me or made me want to share it.

- It doesn't feel that opinionated. Obviously there are opinions in there, but it doesn't feel like we have something important to share here. Case in point, our final point is "Don't reinvent the wheel".

- Just structurally, it feels a little formulaic / SEO-like. 

I still think "docs" as a topic is potentially interesting, but this isn't the right way to tackle it.

andyvan-ph commented:
Did some thinking about what might work and I feel like it needs a format that grounds it real world examples. 

Something like "The X characteristics of great documentation" (bad headline) where it goes:

- Good docs do X
- Explain X
- A good example is Stripe.
- Screenshot of what you're talking about
- Break down why it works / why it's good

ivanagas commented:
Alright, completely rewrote with new frame. 

I looked into going the characteristics and using a bunch of other companies as examples but it is really hard to draw examples from other people's docs. Most docs analysis just says "Stripe is good" which isn't really helpful.

andyvan-ph reviewed (commented):
This is way better than the first attempt, nice one.

Haven't done a deep read just yet, but first thought was could we add a section on using analytics / session replay to see how people use docs?

We could use my recent sidebar re-org as an example of using heatmaps to figure out where people were going, what we should prioritize etc.

andyvan-ph reviewed (commented):
Did a deeper a read and left a few comments. Beyond that, I'm trying to think of other points we should cover in this. Some random thoughts:

- Should we talk about how AI is / will change how people consume documentation?

- This is more of a tactical thing, but I think it would be cool to add a point about how we've learned to hide important info in a detail expand with `(recommended)` on it, and that people are more likely to engage with and read it this way. We could use a heatmap screenshot from the `Install PostHog page` to demonstrate this.

- Add the benefits of making docs easy for anyone to contribute? Maybe this isn't big enough for its own section, though.

- One thing we don't cover, which we 100% should, is the difference between docs and tutorials. They're both important, but they're different things. Would be good explore this a little bit.

- Perhaps we could add a kind of common mistakes section, maybe as a bonus section? This is great, but I feel like a few more tactical things people can take away will give it an extra 5%.

daniloc commented on contents/newsletter/writing-docs.md:
```diff
@@ -0,0 +1,129 @@
+
+When users start with your product, you want them to go from “nothing” to “something” as quickly as possible. The best way to do this is learning by doing. Docs provide the structure to do this.
+
+Start by writing the stupidest, most basic doc to help someone use your product. What would you send to a friend to help them get started with this feature?
```
```suggestion
Start by writing the most basic, obvious doc to help someone use your product. What would you send to a friend to help them get started with this feature?
```

Instead of stupid, how would you feel about "obvious"? I find this nudge can be more clarifying because "the obvious" is more obvious than the stupid, if that makes any sense.

daniloc commented on contents/newsletter/writing-docs.md:
```diff
@@ -0,0 +1,129 @@
+
+For example, our Next.js doc [started](https://github.com/PostHog/posthog.com/pull/1842) as a simple page with one snippet for installing and another for capturing a custom event. Since then, they changed 49 times to the current version with multiple install methods, details on app and pages routers, frequently asked questions, and more.
+
+![Next.js docs](https://res.cloudinary.com/dmukukwp6/image/upload/image_1_6c2f84d260.png)
```
I think side-by-side allows for better comparison, but what if we trimmed out all the sidebar stuff from both specimens?

ivanagas replied:
![nextdocs](https://github.com/user-attachments/assets/5acb0731-98bf-4f7a-bb6c-3f4506136a15)

Better?

ivanagas replied:
The problem with this is that it's not obvious it has changed that much lol

daniloc commented on contents/newsletter/writing-docs.md:
```diff
@@ -0,0 +1,129 @@
+- Use bullet points and numbered lists as these help readers know where they are and create a sense of progress.
+- Add functioning code samples, annotated screenshots, graphics, and even memes. Visuals help keep readers’ attention and provide an alternative way to explain a concept.
+
+Ideally, your writing should have texture and flow. This helps readers effectively find and understand what they are looking for so they can get back to work.
```
I think if the example appeared AHEAD of "texture and flow" it would be really clear.

ivanagas replied:
Going to drop texture and flow as ideas and use less abstract words

daniloc commented on contents/newsletter/writing-docs.md:
```diff
@@ -0,0 +1,129 @@
+
+![Identify](https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2025_04_03_at_12_01_092x_01127d03fc.png)
+
+To focus more on examples, think about “showing” rather than “telling.”
```
These are really help examples and I wonder if you could do with less throat-clearing on the subject before you got to them.

lol it occurs to me I'm giving you your own advice here: the examples are more potent than the abstract concepts

ivanagas replied:
Helpful feedback, I reworked this section to weave the user identification example through the section rather than having so much throat clearing at the start.

daniloc commented:
> Most docs analysis just says "Stripe is good" which isn't really helpful.

I lol'ed, this is true

I like how helpful this piece is. one add-on to take or leave: I think it makes a big difference to have a role model in something like this. Encouraging the reader to think about the docs they like, and why they like them, could provide a bit of aspirational fuel to drive the engine you described here. we could perhaps seed the exercise with our own taste on that point.

Implicitly, one of those models will be us if they're reading this. but we're SO COMPLEX, supporting so many products, some tighter focused examples could be more relevant for the beginner.

ivanagas commented on contents/newsletter/writing-docs.md:
```diff
@@ -0,0 +1,129 @@
+
+Ideally, your writing should have texture and flow. This helps readers effectively find and understand what they are looking for so they can get back to work.
+
+![Texture](https://res.cloudinary.com/dmukukwp6/image/upload/image_3_25886bf103.png)
```
![yes](https://res.cloudinary.com/dmukukwp6/image/upload/badrad_14f62de3dc.png)

better?

ivanagas commented:
@andyvan-ph 
- I think we are starting our AI documentation journey with the wizard and AI focused tools, so I don't have a great, section-worthy take for this yet. If you have an idea, let me know.
- Added a bit on `<details>` component to "Your readers are in a rush"
- I do mention contributing in "Good docs are not written in one day" but maybe that bit needs to be more prominent
- Added a note on tutorials in "Writing great docs is like building a great product."  "3. **Prioritize what really matters.** If you're reading this, you probably shouldn't be writing tutorials. They're useful, but less important than getting started or reference docs. Constantly evaluate what docs are most important to work on so you maximize your impact."
	- Basically my take is that engineers or founders shouldn't be writing tutorials, that is a later stage problem

Some potential sections to add:
1. Common mistakes
2. How to prioritize docs work (would require reworking some other sections, but that might be ok because each are probably a bit long now)

@daniloc 
- Tried making the more good reads some examples of great docs and a tiny bit of why they are great.
